ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committing/ProcessTransactionSubmission.scala

```diff
@@ -208,6 +212,63 @@ private[kvutils] case class ProcessTransactionSubmission(
           buildRejectionLogEntry(RejectionReason.Disputed("DuplicateKey: Contract Key not unique")))
     } yield r
 
+  /** Check that all parties mentioned in a transaction exist. */
+  private def validateParties: Commit[Unit] = {
+
+    def foldPartiesInTx[T](tx: GenTransaction.WithTxValue[NodeId, TContractId], z: T)(f: (T, String) => T): T =
```

Comment:
Looks quite generic. Do you think it belongs in `GenTransaction`?

Reply:
I tend to be conservative (esp. at this stage of my code base awareness) but yes, it could be already moved there if you think it makes sense